WordPress/Classes/Models/BlogSettings+DateAndTimeFormat.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+
+/// In this extension, we implement several nested Enums (and helper setters / getters) aimed at simplifying
+/// the BlogSettings interface for handling writing date and time format properties.
+///
+extension BlogSettings {
+
+    /// Enumerates the Date format settings.
+    ///
+    enum DateFormat: String {
+        case MonthDY = "F j, Y"
+        case YMD = "Y-m-d"
+        case MDY = "m/d/Y"
+        case DMY = "d/m/Y"
+
+        /// Returns the sorted collection of all of the Localized Enum Titles.
+        ///
+        static var allTitles: [String] {
+            return allValues.flatMap { descriptionMap[$0] }
+        }
+
+        /// Returns the sorted collection of all of the possible Enum Values.
+        ///
+        static var allValues: [String] {
+            return [DateFormat.MonthDY.rawValue, DateFormat.YMD.rawValue,
+                    DateFormat.MDY.rawValue, DateFormat.DMY.rawValue]
+        }
+
+        /// Returns the localized description of the current enum value.
+        ///
+        var description: String {
+            return DateFormat.descriptionMap[rawValue]!
+        }
+
+        // MARK: - Private Properties
+
+        fileprivate static let descriptionMap = [
+            MonthDY.rawValue: NSLocalizedString("December 17, 2017", comment: "Only December needs to be translated"),
+            YMD.rawValue: NSLocalizedString("2017-12-17", comment: "No translation required"),
+            MDY.rawValue: NSLocalizedString("12/17/2017", comment: "No translation required"),
+            DMY.rawValue: NSLocalizedString("17/12/2017", comment: "No translation required")
```

Comment:
If the last three don't need translation, why wrap them in `NSLocalizedString`?
As for the first one, we could use a `DateFormatter` with a custom format and get translations for free.

Reply:
Fixed, thanks for the suggestion, I was being a bit robotic on this one.